foundry/foundry.py

```diff
@@ -113,6 +113,11 @@ def load(self, name, download=True, globus=True, verbose=False, metadata=None, *
         -------
             self
         """
+        # handle empty dataset name (was returning all the datasets)
+        if not name:
+            print("load: No dataset name is given")
+            exit()
```

Comment:
instead of `exit()`, we should `raise` an Exception. See https://docs.python.org/3/tutorial/errors.html#raising-exceptions for how to properly raise an Exception. Here is the list of Python Exceptions (you want to pick the one that is the most specific, and makes sense for the issue): https://docs.python.org/3/library/exceptions.html

Reply:
I made a change!